lib/web_ui/lib/src/engine/semantics/semantics.dart

```diff
@@ -1676,6 +1685,42 @@ class SemanticsObject {
     }());
     return result;
   }
+
+  /// Leaves an empty "tombstone" detached from any role managers until such
+  /// time as the focus moves to another DOM element.
+  ///
+  /// A tombstone is used when removing a currently focused element from the DOM
+  /// would otherwise cause some screen readers, such as TalkBack, to move focus
+  /// to the <body> tag, which is not what the user wants.
+  void leaveTombstone() {
+    primaryRole?.dispose();
+
+    final int id = this.id;
+    element.clearChildren();
+    element.setAttribute('aria-hidden', true);
+    element.setAttribute('flt-tombstone', '');
+    element.style.transform = 'translate(-1px, -1px)';
+    element.style.top = '';
+    element.style.left = '';
+    element.style.width = '0px';
+    element.style.height = '0px';
+
+    final DomEventListener blurListener = createDomEventListener((_) {
+      owner._tombstones.remove(id);
```

Comment:
Once that a tombstone is left, it seems to never be re-used, so do we really need to keep a map of `_tombstones` by `id` in the semantics `owner`? I say that once you schedule them for deletion after onBlur, nobody else cares about those objects anymore, right? (so no need to keep track of the Tombstone typedef for it to be eventually deleted, we just let it go onBlur)

Reply:
This is worth a try IMHO. Another aspect to consider here is whether it's even possible to have more than one tombstone at a time?

Consider the following sequence of events:
1. Node1 disposed ==> 1st tombstone.
2. Node2 focused ==> we can safely remove the 1st tombstone now.
3. Node2 disposed ==> 2nd tombstone (but at this point, the 1st tombstone is gone).

If my understanding is correct, we don't need a map of tombstones, we just need the `Tombstone? _lastTombstone` (if at all).